sdk/identity/azure-identity/azure/identity/_credentials/azure_arc.py

```diff
@@ -38,23 +39,32 @@ def __init__(self, **kwargs):
         super(AzureArcCredential, self).__init__()
 
         url = os.environ.get(EnvironmentVariables.IDENTITY_ENDPOINT)
-        if not url:
+        imds = os.environ.get(EnvironmentVariables.IMDS_ENDPOINT)
+        if not (url and imds):
             # Azure Arc managed identity isn't available in this environment
             self._client = None
             return
+
         identity_config = kwargs.pop("_identity_config", None) or {}
         config = _get_configuration()
-        client_args = dict(
-            kwargs,
+
+        self._client = ManagedIdentityClient(
             _identity_config=identity_config,
             policies=_get_policies(config),
             request_factory=functools.partial(_get_request, url),
+            **kwargs
         )
 
-        self._client = ManagedIdentityClient(**client_args)
-
     def get_token(self, *scopes, **kwargs):
         # type: (*str, **Any) -> AccessToken
+        if self._client._identity_config != {}:
```

Comment:
`{}` is falsey, so you can do this:
```suggestion
        if not self._client._identity_config:
```